client/src/pages/Analytics/components/CompareGraph.tsx

```diff
@@ -0,0 +1,288 @@
+  { date: '2023-10-15', amount: 51.4 },
+  { date: '2023-10-16', amount: 59.2 },
+  { date: '2023-10-17', amount: 64.7 },
+  { date: '2023-10-18', amount: 36.8 },
+  { date: '2023-10-19', amount: 48.9 },
+  { date: '2023-10-20', amount: 44.1 },
+  { date: '2023-10-21', amount: 52.4 },
+  { date: '2023-10-22', amount: 61.6 },
+  { date: '2023-10-23', amount: 33.7 },
+  { date: '2023-10-24', amount: 46.8 },
+  { date: '2023-10-25', amount: 38.2 },
+  { date: '2023-10-26', amount: 50.3 },
+  { date: '2023-10-27', amount: 55.4 },
+  { date: '2023-10-28', amount: 69.8 },
+  { date: '2023-10-29', amount: 42.9 },
+  { date: '2023-10-30', amount: 60.1 },
+  { date: '2023-10-31', amount: 37.5 },
+];
+
+const dailyTotals2 = [
+  { date: '2023-09-01', amount: 64.23 },
+  { date: '2023-09-02', amount: 32.65 },
+  { date: '2023-09-03', amount: 89.54 },
+  { date: '2023-09-04', amount: 23.12 },
+  { date: '2023-09-05', amount: 87.57 },
+  { date: '2023-09-06', amount: 23.65 },
+  { date: '2023-09-07', amount: 84.34 },
+  { date: '2023-09-08', amount: 76.23 },
+  { date: '2023-09-09', amount: 15.23 },
+  { date: '2023-09-10', amount: 97.24 },
+  { date: '2023-09-11', amount: 38.75 },
+  { date: '2023-09-12', amount: 83.34 },
+  { date: '2023-09-13', amount: 26.88 },
+  { date: '2023-09-14', amount: 93.35 },
+  { date: '2023-09-15', amount: 88.35 },
+  { date: '2023-09-16', amount: 23.66 },
+  { date: '2023-09-17', amount: 13.25 },
+  { date: '2023-09-18', amount: 64.35 },
+  { date: '2023-09-19', amount: 55.45 },
+  { date: '2023-09-20', amount: 87.24 },
+  { date: '2023-09-21', amount: 12.35 },
+  { date: '2023-09-22', amount: 85.34 },
+  { date: '2023-09-23', amount: 35.24 },
+  { date: '2023-09-24', amount: 86.75 },
+  { date: '2023-09-25', amount: 24.76 },
+  { date: '2023-09-26', amount: 57.76 },
+  { date: '2023-09-27', amount: 68.46 },
+  { date: '2023-09-28', amount: 69.99 },
+  { date: '2023-09-29', amount: 94.46 },
+  { date: '2023-09-30', amount: 35.57 },
+];
+
+function getMonthsArray() {
+  const currDate = new Date();
+  const currMonth = currDate.getMonth();
+  const currYear = currDate.getFullYear();
+  const monthCount = 12;
+  const result = [];
+
+  for (let i = 1; i <= monthCount; i++) {
+    const month = (currMonth - i + monthCount) % monthCount;
+    const year = month < currMonth ? currYear : currYear - 1;
+    const monthString = new Date(year, month).toLocaleString('default', {
+      month: 'short',
+    });
+
+    result.push({
+      month: month,
+      year: year,
+      text: `${monthString} ${year}`,
+    });
+  }
+
+  return result;
+}
+
+/**
+ * Comparison graph component
+ *
+ * @returns {JSX.Element} - comparison graph
+ */
+export const CompareGraph: React.FC = () => {
+  const monthsArray = getMonthsArray();
+  const [month1Index, setMonth1Index] = useState(0);
+  const [month2Index, setMonth2Index] = useState(1);
+  const [data1, setData1] = useState(dailyTotals1);
+  const [data2, setData2] = useState(dailyTotals2);
```

Comment:
i feel like we should find better names to distinguish between the two instead of using numbers

same goes for `entry1` `entry2`

might come across as bad naming convention in the final review w cody